### PR TITLE
Adds support for unit tests.

### LINKED ISSUE #11: moduleForComponent implementation
Hi,

Trying to test components, missing the ember-cli `moduleForComponent` implementation though. Is there a way to test components with ember-cli-yadda or do I have to extend the plugin to add the module?

Cheers,
Michael

### PR DESCRIPTION
**Pull request by mschinis**
Closes #11 

**NOTE** Unit test support only support qunit for now. I don't use mocha, so I didn't get the chance to add mocha support yet.

Will be creating another PR with mocha support, after I look into the syntax!

Also updated the documentation to reflect the new updates

### CODE REVIEW

**Comment by albertjan** [comment]
Thanks! Looking into it now :+1:

File: index.js [hunk 1]
```diff
@@ -46,10 +54,11 @@ FeatureParser.prototype.getTestFrameworkImport = function() {
     case 'mocha':
       return 'import { afterEach, beforeEach, describe, it } from \'mocha\';';
     default: // qunit
-      return 'import { module, test } from \'qunit\';';
+      return 'import { moduleFor, moduleForComponent, test } from \'ember-qunit\';\
+              import {module} from \'qunit\';';
```
Comment (albertjan) [inline index.js]: please update the whitespace for this line to `import { module } from \'qunit\'`.

File: index.js [hunk 2]
```diff
@@ -73,25 +82,51 @@ FeatureParser.prototype.getTestFeature = function() {
...
-        "      this.application = startApp();",
-        "    },",
-        "    afterEach: function() {",
-        "      Ember.run(this.application, 'destroy');",
-        "    }",
-        "  });",
+      if(unitModule && unitModule.indexOf('component') === 0){
+        testFeature = [
+          "function testFeature(feature) {",
+          "  moduleForComponent('"+fileName+"', `Feature: ${feature.title}`,{",
```
Comment (albertjan) [inline index.js]: and add some air here too :smile:

File: index.js [hunk 3]
```diff
@@ -73,25 +82,51 @@ FeatureParser.prototype.getTestFeature = function() {
...
+          "    test(`Scenario: ${scenario.title}`, function(assert) {",
+          "      expect(scenario.steps.length);",
+          "      let self = this;",
+          "      return new Ember.RSVP.Promise(function(resolve){ yadda.Yadda(library.default(assert), self).yadda(scenario.steps, { ctx: {} }, resolve); });",
+          "    });",
+          "  });",
+          "};"
+        ];
+      }else if(unitModule && unitModule.indexOf('model') === 0){
+        // TODO: Implement unit test for models
```
Comment (albertjan) [inline index.js]: maybe a good idea to write something friendly to the command line when someone uses it :smile:

File: index.js [hunk 4]
```diff
@@ -1,6 +1,7 @@
 var yadda = require('yadda');
 var Filter = require('broccoli-filter');
 var path = require('path');
+var inflected = require( 'inflected' );
```
Comment (albertjan) [inline index.js]: too much air :neckbeard: sorry

File: index.js [hunk 5]
```diff
@@ -46,10 +54,11 @@ FeatureParser.prototype.getTestFrameworkImport = function() {
     case 'mocha':
       return 'import { afterEach, beforeEach, describe, it } from \'mocha\';';
     default: // qunit
-      return 'import { module, test } from \'qunit\';';
+      return 'import { moduleFor, moduleForComponent, test } from \'ember-qunit\';\
+              import {module} from \'qunit\';';
   }
 };
-FeatureParser.prototype.getTestFeature = function() {
+FeatureParser.prototype.getTestFeature = function(unitModule,fileName ) {
```
Comment (albertjan) [inline index.js]: here's a bit to much as well ... I'll stop nitpicking now I promise.

File: blueprints/qunit/ember-cli-yadda/files/tests/unit/steps/steps.js [hunk 6]
```diff
@@ -0,0 +1,14 @@
+import yadda from '../../helpers/yadda';
+
+export default function(assert) {
+  return yadda.localisation.English.library()
+    .given('I type "Ember g feature make-feature"', (next) => {
+      visit('/');
```
Comment (albertjan) [inline blueprints/qunit/ember-cli-yadda/files/tests/unit/steps/steps.js]: `visit` is not available in the unittest world so it generates a failing test, which none of the other blueprints in ember do so I think we shouldn't either.

File: blueprints/qunit/ember-cli-yadda/files/tests/unit/steps/steps.js [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+      assert.ok(true, this.step);
+      andThen(() => next());
```
Comment (albertjan) [inline blueprints/qunit/ember-cli-yadda/files/tests/unit/steps/steps.js]: `andThen` also doesn't exist

File: index.js (hunk 1 above)
Comment (mschinis) [inline index.js]: Do you need me to remove leading whitespace on this line as well?

Comment (albertjan) [inline index.js]: No thats ok I like that they're lined up.

**Comment by mschinis** [comment]
Done, sorry for the delay. :D

**Comment by albertjan** [comment]
Hey man it's christmas! No worries, have fun!

**Comment by albertjan** [comment]
I pull it in by hand because there was a merge conflit in README :) Thanks for the contribution.

**Comment by albertjan** [comment]
I'll do a update of npm soon when I've played with it a little more.

**Comment by mschinis** [comment]
Cool thanks!

**Pull request closed without merging**